fix: Update search shortcut to open at position for no selection

KattMingMing opened the pull request:
Resolves https://github.com/sourcegraph/sourcegraph-jetbrains/issues/3 and https://github.com/sourcegraph/sourcegraph-jetbrains/issues/2

slimsag commented on README.md:
```diff
@@ -73,6 +73,7 @@ Please file an issue: https://github.com/sourcegraph/sourcegraph-jetbrains/issue
 
 ## Version History
 
+- v1.1.2 - Updated the search action to trigger the open action when no text is selected.
```
As a user I find this behavior confusing. If I activate the search shortcut and there is no selection, I expect it to do nothing. Opening the file is confusing because I am expecting search results

KattMingMing replied:
I see. How about if it opened to the repository with the search input active instead of opening to the file position with the current cursor selected?

KattMingMing commented:
@slimsag - I updated the PR so that the search action properly checks for a null selection and also searches as a string literal and escapes double quotes to prevent search syntax errors. (For example, searching for `foo()` and `bar("hello"` would not be valid queries without the change.)

I removed the ability to search without a selection

slimsag commented on src/Search.java:
```diff
@@ -64,7 +60,7 @@ public void actionPerformed(AnActionEvent e) {
                     + "&utm_product_name=" + URLEncoder.encode(productName, "UTF-8")
                     + "&utm_product_version=" + URLEncoder.encode(productVersion, "UTF-8")
-                    + "&search=" + URLEncoder.encode(q, "UTF-8");
+                    + "&search=" + URLEncoder.encode(String.format("\"%s\"", q.replaceAll("\"", "\\\\\"")), "UTF-8");
```
What is this about?
